### DISCUSSION TITLE
C812 missing trailing comma

### DISCUSSION THREAD

**Discussion by cidrblock** [ansible/ansible-navigator#712] [Linters, linting, formatting, pre-commit, CI]
```
command, additional_params = _generate_command(
        exec_command=cmd_test_data.command, exec_shell=cmd_test_data.shell
    )
```

https://www.python.org/dev/peps/pep-0008/#id27

is what triggered it, I don't really have an issue with data structures that might grow

this was a good read

https://stackoverflow.com/questions/12087742/should-i-add-a-trailing-comma-after-the-last-argument-in-a-function-call

It feels odd and out of place, maybe because it's my first time seeing it

**Comment by cidrblock** [comment]
although, once black ran and the comma was added

```
command, pass_through_args = _generate_command(
            exec_command=self._args.exec_command,
            exec_shell=self._args.exec_shell,
        )
```

this does look better

**Comment by cidrblock** [comment]
another nice example of where it help:

before:
```
diff = "\n".join(
                difflib.unified_diff(expected_output, received_output, "expected", "received")
            )
```

after a comma and black

```
diff = "\n".join(
                difflib.unified_diff(
                    expected_output,
                    received_output,
                    "expected",
                    "received",
                )
            )
```
I find readability went up here

**Reply by webknjaz** [reply]
Hm... The generic "trailing comma is better for `git diff`" is a well-known problem. What black did with the indentation is a bit weird, though.

Note that we do have an autofixer that deals with this specific case in the config, too bad it's disabled now — https://github.com/ansible/ansible-navigator/blob/272976c/.pre-commit-config.yaml#L4-L15.
